SUASSystem/SUASSystem/gcs.py

```diff
@@ -48,24 +50,25 @@ def gcs_process(sda_status, img_proc_status, interop_client_array, targets_to_su
         location_log.append(current_location_json)
 
         vehicle_state_data[0] = SUASSystem.get_vehicle_state(vehicle, GCSSettings.MSL_ALT)
+
         if len(interop_client_array) != 0:
             interop_client_array[0].post_telemetry(current_location, vehicle_state_data[0].get_direction())
             mission_information_data[0] = get_mission_json(interop_client_array[0].get_active_mission(), interop_client_array[0].get_obstacles())
 
-        if sda_status.value.lower() == "connected":
-            """if (vehicle.location.global_relative_frame.alt * 3.28084) > GCSSettings.SDA_MIN_ALT and (vehicle.mode.name == "GUIDED" or vehicle.mode.name == "AUTO"):
+        if (vehicle.location.global_relative_frame.alt * 3.28084) > GCSSettings.SDA_MIN_ALT:#and sda_status.value.lower() == "connected":
+            print("Condition is TRUE")
+            if (UAV_status.value == "GUIDED"):
+                sda_avoid_feet_height = Location(sda_avoid_coords[0].get_lat(), sda_avoid_coords[0].get_lon(), sda_avoid_coords[0].get_alt()*3.28084)
+                print('sda avoid feet height')
+                print(sda_avoid_feet_height)
                 log("root", "Avoiding obstacles...")
-                vehicle.mode = VehicleMode("GUIDED")
-                guided_waypoint_location = sda_avoid_coords[0][0]
-                vehicle.simple_goto(guided_waypoint_location.as_global_relative_frame())
+                vehicle.mode = dronekit.VehicleMode("GUIDED")
+                vehicle.simple_goto(sda_avoid_coords[0].as_global_relative_frame())
+            if UAV_status.value == 'AUTO' and vehicle.mode.name != "AUTO":
```

Comment:
There should be a new line above a new if statements. You do this in line 57 of this file, but not here.